sql/plan/show_create_table.go

```diff
@@ -0,0 +1,124 @@
+package plan
+
+import (
+	"fmt"
+	"strings"
+)
+
+import (
+	"gopkg.in/src-d/go-errors.v1"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+)
+
+var ErrTableNotFound = errors.NewKind("Table `%s` not found")
+
+// ShowCreateTable is a node that shows the CREATE TABLE statement for a table.
+type ShowCreateTable struct {
+	Database sql.Database
+	Table    string
+	Registry *sql.IndexRegistry
+}
+
+// Schema implements the Node interface.
+func (n *ShowCreateTable) Schema() sql.Schema {
+	return sql.Schema{
+		&sql.Column{Name: "Table", Type: sql.Text, Nullable: false},
+		&sql.Column{Name: "Create Table", Type: sql.Text, Nullable: false},
+	}
+}
+
+// TransformExpressionsUp implements the Transformable interface.
+func (n *ShowCreateTable) TransformExpressionsUp(f sql.TransformExprFunc) (sql.Node, error) {
+	return n, nil
+}
+
+// TransformUp implements the Transformable interface.
+func (n *ShowCreateTable) TransformUp(f sql.TransformNodeFunc) (sql.Node, error) {
+	return f(NewShowCreateTable(n.Database, n.Table, n.Registry))
+}
+
+// RowIter implements the Node interface.
+func (n *ShowCreateTable) RowIter(*sql.Context) (sql.RowIter, error) {
+	return &showCreateTablesIter{
+		db:    n.Database,
+		table: n.Table,
+	}, nil
+}
+
+// String implements the Stringer interface.
+func (n *ShowCreateTable) String() string {
+	return fmt.Sprintf("ShowCreateTable(%s)", n.Table)
+}
+
+type createTableStmt struct {
+	colName string
+	colType sql.Type
+}
+
+type showCreateTablesIter struct {
+	db    sql.Database
+	table string
+
+	createStmt *createTableStmt
+}
+
+func (i *showCreateTablesIter) Next() (sql.Row, error) {
+	table := i.db.Tables()[i.table]
+
+	if table == nil {
+		return nil, ErrTableNotFound.New(table)
+	}
+
+	schema := table.Schema()
+	colCreateStatements := make([]string, len(schema), len(schema))
+	// Statement creation parts for each column
+	for indx, col := range schema {
+		createStmtPart := fmt.Sprintf("`%s` %s", col.Name, col.Type.Type())
+		if col.Default != nil {
+			createStmtPart = fmt.Sprintf("%s DEFAULT %v", createStmtPart, col.Default)
+		}
+
+		if !col.Nullable {
+			createStmtPart = fmt.Sprintf("%sNOT NULL", createStmtPart)
+		}
+
+		if indx != len(schema)-1 {
```

Comment:
I believe we don't need this. At the end you can call: `strings.Join(colCreateStatements, ",\n")`